src/Monolog/Handler.php

```diff
@@ -57,6 +57,15 @@ protected function write(array $record): void
             $scope->setExtra('monolog.channel', $record['channel']);
             $scope->setExtra('monolog.level', $record['level_name']);
 
+            foreach ($record['context'] as $key => $value) {
```

Comment:
@glensc Hi i have two ideas :)

The first one is to maybe just add logger context as  `$scope->setExtra('monolog.context', $record['context']);` and then the user can freely filter or anonymize this array later in `before_send` hook if he needs it.

Sec idea... maybe for now just add context to hint `$hint->extra = ['monolog.context', $record['context']];` so the user can then manually filter & add data he needs to event in `before_send` hook.

Just "my two cents".

Reply:
@Jean85 @stayallive opinions about the first point? I'm kind of undecided if we should put everything under the extra context or if as, suggested, it would be wiser to group everything under monolog.context

Reply:
Maybe the second option, a dedicated context, would be better? That way we could avoid situation where we accidentally override user-submitted data.

Reply:
Done!

Reply:
@ste93cry if you inow how to rebase and force push, why don't you already cleanup the history? like why have commit stuff, revert stuff? just squash so that it's a commit that adds stuff with the reverted stuff in same commit?

Reply:
Keeping the history of commits in the PR is sometimes useul to keep track of the process that led to the final version of the code to review, giving more context to the reviewer. Besides that, squashing is done when merging into the target branch so it doesn't really matter how many commits there are here, the history of the repository will be clean anyway :wink:

_Funny fact: you were the one advocating for commits to be as atomic as possible so that it's possible to revert them without losing other unrelated changes_

Reply:
That completely changes the story if the pull request is squash merged. I wish github/gitlab present it to reviewers that squash merge is used, then could just omit any commit-related quality checks.

as for the atomic part, for the purposes of merging cleaner history, I would exclude (with rebase) changes from merge that are removed rather than merging revert commits. as everything before merge can be rewritten. if you want to preserve historic decisions, then maybe start with new pull request leave old one for posterity.

so far both github/gitlab still don't present well previous versions of pushes. gitlab at least has merge request versions, but in github you have to dig through timeline find the "has force pushed" and then can find previous versions.